sos/plugins/juju.py

```diff
@@ -76,37 +86,40 @@ def export_mongodb(self):
                 suggest_filename="{}.json".format(collection))
 
     def setup(self):
-        self.add_copy_spec([
-            "/var/lib/juju"
-        ])
         limit = self.get_option("log_size")
         self.add_copy_spec_limit("/var/log/upstart/juju-db.log",
                                  sizelimit=limit)
         self.add_copy_spec_limit("/var/log/upstart/juju-db.log.1",
                                  sizelimit=limit)
         if not self.get_option("all_logs"):
             # Capture the last bit of all files
-            for filename in os.listdir("/var/log/juju/"):
-                if filename.endswith(".log"):
-                    fullname = "/var/log/juju/" + filename
-                    self.add_copy_spec_limit(fullname, sizelimit=limit)
-            # Do just the all-machines from juju local
-            self.add_copy_spec_limit("/var/log/juju-*/all-machines.log",
-                                     sizelimit=limit)
+            if(os.path.isdir("/var/log/juju/")):
+                for filename in os.listdir("/var/log/juju/"):
+                    if filename.endswith(".log"):
+                        fullname = "/var/log/juju/" + filename
+                        self.add_copy_spec_limit(fullname, sizelimit=limit)
             self.add_cmd_output('ls -alRh /var/log/juju*')
+            self.add_cmd_output('ls -alRh /var/lib/juju/*')
+
         else:
             self.add_copy_spec([
                 "/var/log/juju",
-                "/var/log/juju-*"
+                "/var/log/juju-*",
+                "/var/lib/juju"
             ])
 
-        self.add_cmd_output([
-            "juju -v status",
-            "juju -v get-constraints"
-        ])
-
-        for service in self.get_deployed_services():
-            self.add_cmd_output("juju get {}".format(service))
+        if(os.path.exists("/usr/bin/juju")):
```

Comment:
This doesn't need to be in an if block (and generally we do not check for command presence in sos unless it is absolutely necessary). Just attempt the `add_cmd_output()` and if the command doesn't exist it won't be collected. Similarly with the `get_deployed_services()` stuff - just have the function return an empty list if it can't run the command and the loop bodies disappear.